moztelemetry/dataset.py

```diff
@@ -70,6 +71,9 @@ class Dataset:
         schema = ['submissionDate', 'docType', 'platform']
 
         records = Dataset(bucket, schema) \\
+            .select('clientId',
+                    os_name='environment.system.os.name',
```

Comment:
Might be worthwhile adding some more examples showcasing things that we couldn't do before, like accessing the content of an array or something.